NewsFeed/views/CCBRNewsViewController.m

```diff
@@ -167,6 +170,15 @@ - (void)collectionView:(UICollectionView *)collectionView
 didSelectItemAtIndexPath:(NSIndexPath *)indexPath {
     [self.dispatcher showNewsWithDataSource:self.viewModel.dataSource
                                  startIndex:indexPath.row];
+    [self.viewModel logClickEventForItemAtIndex:indexPath.row];
+}
+
+- (void)collectionView:(UICollectionView *)collectionView willDisplayCell:(UICollectionViewCell *)cell forItemAtIndexPath:(NSIndexPath *)indexPath {
+    [self.viewModel logImpresstionEventForItemAtIndex:indexPath.row];
+    
+    if (self.viewModel.endIndex == (indexPath.row)) {
```

Comment:
This logic should be moved to the ViewModel. View or ViewController should be be as dump as possible.